src/neo/Ledger/Blockchain.cs

```diff
@@ -433,6 +433,8 @@ private void OnParallelVerified(ParallelVerified parallelVerified)
             {
                 if (View.ContainsTransaction(parallelVerified.Transaction.Hash))
                     reason = RelayResultReason.AlreadyExists;
+                else if (!parallelVerified.Transaction.VerifyBalance(currentSnapshot, MemPool.SendersFeeMonitor.GetSenderFee(parallelVerified.Transaction.Sender)))
```

Comment:
How about call `VerifyForEachBlock` for a full check here? I think during parallel verify witness, a new block may persist, and snapshot will change to a new one, so the check in `OnNewTransactoin` is not reliable, we need to check again in `OnParallelVerified`.

Reply:
@eryeer Feel free to change it, check the balance twice maybe have no sense.

Reply:
I think we should call `VerifyForEachBlock()`, instead of call `VerifyBalance()` only.

Reply:
Done, we should test the speed, I think that maybe the parallel doesn't worth it

Reply:
@cloud8little could your team make a speed test with and without Parallel verification? (with these changes)

Reply:
I think so too.

Reply:
@cloud8little you can use https://github.com/neo-project/neo/pull/1429 for benchmarks between both solutions.

Reply:
@shargon ok, we will do it.

Reply:
@shargon It takes some time to deploy a current version of the test environment, give us some days to test them.